pinot-spi/src/main/java/org/apache/pinot/spi/config/table/UpsertConfig.java

```diff
@@ -44,16 +44,16 @@ public enum Strategy {
   private final Mode _mode;
 
   @JsonPropertyDescription("Function to hash the primary key.")
-  private final HashFunction _hashFunction;
+  private HashFunction _hashFunction;
 
   @JsonPropertyDescription("Partial update strategies.")
-  private final Map<String, Strategy> _partialUpsertStrategies;
+  private Map<String, Strategy> _partialUpsertStrategies;
 
   @JsonPropertyDescription("default upsert strategy for partial mode")
-  private final Strategy _defaultPartialUpsertStrategy;
+  private Strategy _defaultPartialUpsertStrategy;
 
   @JsonPropertyDescription("Column for upsert comparison, default to time column")
-  private final String _comparisonColumn;
+  private String _comparisonColumn;
 
   @JsonCreator
```

Comment:
Remove this annotation and add `@Deprecated`

Reply:
Thanks, addressed the comment.